src/dos/dos_locale.cpp

```diff
@@ -1384,28 +1397,28 @@ static void load_keyboard_layout()
 	bool using_detected     = false; // if layout list is autodetected
 	bool code_page_supplied = false; // if code page given in the parameter
 
-	if (config.keyboard_str == "auto") {
+	const auto tokens = split(config.keyboard_str);
+	if (tokens.size() > 2) {
+		LOG_WARNING("LOCALE: Invalid 'keyboard_layout' setting '%s', using 'auto'",
+		            config.keyboard_str.c_str());
```

Comment:
There's a standard error string template for this (`Invalid '%s' setting, using '%s'`) in `support.cpp` somewhere I believe.